src/openms/include/OpenMS/CHEMISTRY/ModificationsDB.h

```diff
@@ -119,10 +113,9 @@ namespace OpenMS
 
        If more than one matching modification is found, the first one is returned with a warning.
 
-       @throw Exception::ElementNotFound if no modification named @p mod_name exists (via searchModifications())
-       @throw Exception::InvalidValue if no matching modification exists
+       Returns nullptr if no modification named @p mod_name exists (via searchModifications()) or if no matching modification exists
     */
-    const ResidueModification& getModification(const String& mod_name, const String& residue = "", ResidueModification::TermSpecificity term_spec = ResidueModification::NUMBER_OF_TERM_SPECIFICITY) const;
+    const ResidueModification* getModification(const String& mod_name, const String& residue = "", ResidueModification::TermSpecificity term_spec = ResidueModification::NUMBER_OF_TERM_SPECIFICITY) const;
```

Comment:
@timosachsenberg hmm - so doesnt searchModifications still throw, see here:

https://github.com/timosachsenberg/OpenMS/blob/e1d796d5763446630e3c9f8582144505c6684144/src/openms/source/CHEMISTRY/ModificationsDB.cpp#L119-L123

Reply:
Yes. I will add to this PR. So far only ResidueDB is kind of fixed. If it calls moddb there are still problems

Reply:
In this case I can simply print a warning return without throwing.
Need to make sure the calling code does not use try / catch for any semantics though